src/goto-programs/interpreter.cpp

```diff
@@ -292,7 +292,10 @@ void interpretert::step()
   case RETURN:
     trace_step.type=goto_trace_stept::typet::FUNCTION_RETURN;
     if(call_stack.empty())
-      throw "RETURN without call"; // NOLINT(readability/throw)
+    {
+      throw analysis_exceptiont(
+        "RETURN without call"); // NOLINT(readability/throw)
```

Comment:
I think this is an INVARIANT as if the interpreter is working as it should, then this should never happen.

Reply:
I was thinking this could happen in a nonsensical goto program that has returns in unexpected places, such as `__CPROVER_initialize`. Is that wrong?

Reply:
If I understand the code correctly (and it is possible I don't) then the INVARIANT is that the number of RETURNS executed should never exceed the number of functions started.  So this should always hold.